zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationPreconditions.java

```diff
@@ -852,7 +854,18 @@ public static void requireNoConcurrentCommand(
     // or
     // An active sequence flow indicates a concurrent command. It is created when taking a
     // sequence flow and writing an ACTIVATE command for the next element.
-    if (eventTrigger != null || elementInstance.getActiveSequenceFlows() > 0) {
+    // We allow migrating joining parallel gateway with at least one incoming sequence flow is taken
+    final long activeFlowsToNonGateway =
+        elementInstance.getActiveSequenceFlowIds().stream()
```

Comment:
❌ We still cannot always trust the `.getActiveSequenceFlowIds` value.

Instead we should do something like this:

```
if (eventTrigger) {
  reject
} else if (elementInstance.activeSequenceFlows > 0) {
  // check if there are active sequence flows to a gateway
  // and whether there are fewer then the generally active sequence flows in this element instance
  if (state.getNumberOfTakenSequenceFlows() < elementInstance.activeSequenceFlows) {
    // there are sequence flows active that don't flow a parallel/inclusive gateway
    reject
  }
}
```

Reply:
@korthout how do we know if the count from `elementInstance.activeSequenceFlows` is complete? E.g. some elements are activated after 8.6 and stored active sequence flow count but it can still be less than the `state.getNumberOfTakenSequenceFlows()` which is wrong. I think, retrieving `elementInstance.getActiveSequenceFlowIds()` and retaining the ones that are not part of `elementInstanceState.getTakenSequenceFlows()` is the more proper way, right?